core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -107,11 +107,15 @@ namespace Impl {
 namespace {
 
 __global__ void query_cuda_kernel_arch(int *d_arch) {
+#if defined(KOKKOS_COMPILER_NVHPC) && defined(_NVHPC_CUDA)
```

Comment:
Isn't 
```suggestion
#ifdef _NVHPC_CUDA
```
sufficient?

Reply:
I suppose so.  Are you asking me to change?

Reply:
We are not using `_NVHPC_CUDA` anywhere else if I'm not mistaken. I think we should only be consistent.

Reply:
That is the first time we use it indeed.
>  I think we should only be consistent.

No one is going to argue with that.  I don't understand what you mean though.  Are you saying you would like me to apply your suggestion?  (I have no objection)

Reply:
Yeah, I would prefer to change it (to be closer to what we do below for CUDA without NVHPC).